src/stacks/spa_pipeline_stack.ts

```diff
@@ -30,10 +34,12 @@ export default class SpaPipelineStack extends BasePipelineStack {
       subDomain = isProduction(appEnvironment)
         ? appName
         : this.conventions.eqn("dash"),
+      domainName,
+      cloudFrontWithDomainSSL,
     } = props;
 
     const bucketWebsite = new Bucket(this, "siteBucket", {
-      bucketName: subDomain.toLowerCase(),
+      bucketName: `${appName.toLowerCase() + appEnvironment.toLowerCase()}`, // i dont think subDomain should be set to the app name and environment. best to stick with www api prod test stag. Then name the bucket after the app and the environment
```

Comment:
This is tricky - there are 2 different scenarios here (also keep in mind that S3 buckets must be unique Globally):

1) S3 bucket website - the bucket name is part of the full domain name.  That's why I was using <app>-<env>.  (eq-staging.s3-website.awsamazon.com).  I think that's a fair default - there is a conflict, they can always pass in a custom subdomain.
2) Cloudfront website - the bucket name doesn't need to match the domain in anyway.  Having said that, it should clearly communicate what app/env/purpose that bucket is for, as well as incorporate some random element to it to avoid collisions.

Reply:
1. eq-staging.s3-website.awsamazon.com This only works if the environment is not production. I agree with the naming convention that you mentioned. But if its a prod env you get eq.s3-website.awsamazon.com, you would if you didn't get an error about eq only being two letters. 

const {
      subDomain = isProduction(appEnvironment)
        ? appName <<<< production env bucket name will only be appName no env or purpose.  
        : this.conventions.eqn("dash"), <<<< if not production then you get a bucket name with app and env. 
    } = props;

2. I think thats my fault during my learning trying to simplify things to first get a grasp then go nuts.